src/masternodes/masternodes.cpp

```diff
@@ -104,6 +104,52 @@ CAmount GetProposalCreationFee(int, const CCustomCSView &view, const CCreateProp
     return -1;
 }
 
+CKeyID GetMNKeyOrDefaultFromDestination(const CTxDestination &dest)
+{
+    switch(dest.index()) {
+        case PKHashType:
+        case WitV0KeyHashType:
+            return CKeyID::FromOrDefaultDestination(dest);
+        default:
+            return {};
+    }
+}
+
+CKeyID GetRewardKeyOrDefaultFromDestination(const CTxDestination &dest)
+{
+    switch(dest.index()) {
+        case PKHashType:
+        case ScriptHashType:
+        case WitV0KeyHashType:
+            return CKeyID::FromOrDefaultDestination(dest);
+        default:
+            return {};
+    }
+}
+
+CTxDestination GetMNDestinationOrDefaultFromKey(const char type, const CKeyID &keyId)
+{
+    switch (type) {
+        case PKHashType:
+        case WitV0KeyHashType:
+            return FromOrDefaultKeyIDToDestination(type, keyId);
+        default:
+            return CTxDestination(CNoDestination());
+    }
+}
+
+CTxDestination GetRewardDestinationOrDefaultFromKey(const char type, const CKeyID &keyId)
+{
+    switch (type) {
+        case PKHashType:
+        case ScriptHashType:
+        case WitV0KeyHashType:
+            return FromOrDefaultKeyIDToDestination(type, keyId);
+        default:
+            return CTxDestination(CNoDestination());
+    }
+}
+
```

Comment:
How about we replace this entire chunk with a bit flaggable enum type? Then use this as optional filter to `CKeyId::TryFromDestination` directly?

Reply:
Resolved in commit 787927e20